LimeSurvey/LimeSurvey
# MW-446: flow account and instance existing

To test, one needs to make sure that the

tmp/templates

folder exists and there are lss files inside. (you can export a survey into lss and copy it to the desired location if needed)

Then, encode the filename (without the path) into base64 and, while being logged out, load the

http://ls-ce/index.php?templatetoken=< encoded filename >&r=admin/authentication/sa/login

path. For example:

http://ls-ce/index.php?templatetoken=bGltZXN1cnZleV9zdXJ2ZXlfOTc5NTczLmxzcw==&r=admin/authentication/sa/login

where bGltZXN1cnZleV9zdXJ2ZXlfOTc5NTczLmxzcw== is the token of limesurvey_survey_979573.lss.

The page will redirect you to the login page where you could log in.

After logging in, we expect to be redirected to the survey page and a popup being shown asking whether you want to install the template.

If you leave this unanswered and reload the page, the confirm dialog should still be displayed.

If you click:

- outside of the popup content
- the X of the dialog
- No

then the popup should be dismissed and if you reload the page, the popup should no longer be displayed.

Otherwise, if you click on Yes, then a peview should be shown (in the context of this PR this is an empty popup, it's a known issue) and if you close it, then the grid should be refreshed with the imported template.
